lib/Doctrine/Common/Lexer/AbstractLexer.php

```diff
@@ -203,12 +191,9 @@ public function skipUntil($type)
     /**
      * Checks if given value is identical to the given token.
      *
-     * @param mixed      $value
-     * @param int|string $token
-     *
      * @return bool
      */
-    public function isA($value, $token)
+    public function isA(mixed $value, int|string $token)
     {
         return $this->getType($value) === $token;
```

Comment:
Do we even expect something different than a string here? My preference would be to change the declaration of `isA()`.

Reply:
static analysis has nothing to say because phpstan runs at level 7, not at max level (i.e. 9). Until level 8, `mixed` behaves like Typescript's `any`: everything is valid with it. It is only at level 9 that it is treated like Typescript's `unknown`